src/services/WolfDen.API/Controllers/LeaveManagement/LeaveRequestController.cs

```diff
@@ -16,5 +23,48 @@ public async Task<LeaveRequestHistoryResponseDto> GetLeaveRequestHistory([FromQu
         { 
             return await _mediator.Send(query,cancellationToken); 
         }
+
+        [HttpPost]
+        public async Task<bool> ApplyLeaveRequest( [FromBody] AddLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            return await _mediator.Send(command, cancellationToken);
+        }
+
+        [HttpPatch("revove-leave")]
+        public async Task<bool> RevokeLeave([FromBody] RevokeLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            return await _mediator.Send(command, cancellationToken);
+        }
+
+        [HttpGet("subordinate-leave-requets{id}/{status}")]
+        public async Task<List<SubordinateLeaveDto>> GetSubordinatesLeaveRequest(int id, LeaveRequestStatus status,CancellationToken cancellationToken)
+        {
+            GetSubordinateLeaveQuery query= new GetSubordinateLeaveQuery();
+            query.Id =id;
+            query.StatusId = status;
+
+            return await _mediator.Send(query,cancellationToken);
+        }
+
+        [HttpPatch("subordinate-leave-requets/{id}")]
+        public async Task<bool> ApproveOrRejectLeave(int id , [FromBody] ApproveOrRejectLeaveRequestCommand command,CancellationToken cancellationToken)
+        {
+            command.SuperiorId = id;   
+            return await _mediator.Send(command,cancellationToken);
+        }
+
+        [HttpPatch("edit-leave/{id}")]
```

Comment:
use put for updating if there is more than 2 column changes

Reply:
done